applications/GeoMechanicsApplication/custom_elements/interface_element.cpp

```diff
@@ -323,12 +333,71 @@ std::vector<Vector> InterfaceElement::CalculateTractionsAtIntegrationPoints(cons
 void InterfaceElement::ApplyRotationToBMatrix(Matrix& rBMatrix, const Matrix& rRotationMatrix) const
 {
     const auto dim = GetGeometry().WorkingSpaceDimension();
-    for (int i = 0; i + dim <= rBMatrix.size2(); i += dim) {
+    for (auto i = std::size_t{0}; i + dim <= rBMatrix.size2(); i += dim) {
         auto sub_matrix = subrange(rBMatrix, 0, rBMatrix.size1(), i, i + dim);
         sub_matrix.assign(Matrix{prod(sub_matrix, trans(rRotationMatrix))});
     }
 }
 
+void InterfaceElement::InterpolateNodalStressesToIntegrationPointTractions(
+    const std::vector<std::optional<Vector>>& interface_nodal_cauchy_stresses) const
+{
+    // interpolate nodal stresses on a chosen side
+    auto&      r_interface_geometry = GetGeometry();
+    const auto number_of_nodes_on_side =
+        GeometryUtilities::GetNodeIdsFromGeometry(r_interface_geometry).size() / 2;
```

Comment:
If we only want to determine how many nodes each side has, I don't see why we need to extract the node IDs, look at its total size and divide that by 2. Can't we simply ask for the number of points (i.e. nodes) the geometry has?
```suggestion
    const auto number_of_nodes_on_side =
        r_interface_geometry.PointsNumber() / 2;
```

Reply:
done